src/test_utils.rs

```diff
@@ -107,3 +107,54 @@ pub fn open_gpkg_for_update(path: &Path) -> (TempPath, Dataset) {
     .unwrap();
     (temp_path, ds)
 }
+
+/// Assert numerical difference between two expressions is less than
+/// 64-bit machine epsilon or a specified epsilon.
+///
+/// # Examples:
+/// ```rust, no_run
+/// use gdal::assert_near;
+/// use std::f64::consts::{PI, E};
+/// assert_near!(PI / E, 1.1557273497909217);
+/// // with specified epsilon
+/// assert_near!(PI / E, 1.15572734, epsilon = 1e-8);
+/// ```
+#[macro_export]
+macro_rules! assert_near {
+    ($left:expr, $right:expr) => {
+        assert_near!($left, $right, epsilon = f64::EPSILON)
+    };
+    ($left:expr, $right:expr, epsilon = $ep:expr) => {
+        assert!(
+            ($left - $right).abs() < $ep,
+            "|{} - {}| = {} is greater than epsilon {:.4e}",
+            $left,
+            $right,
+            ($left - $right).abs(),
+            $ep
+        )
+    };
+    ($left:expr, $right:expr, epsilon = $ep:expr, field = $field:expr) => {
+        assert!(
+            ($left - $right).abs() < $ep,
+            "field {}: |{} - {}| = {} is greater than epsilon {:.4e}",
+            $field,
+            $left,
+            $right,
+            ($left - $right).abs(),
+            $ep
+        )
+    };
+    // Pseudo-specialization
+    (StatisticsAll, $left:expr, $right:expr, epsilon = $ep:expr) => {
```

Comment:
Nit: not really a fan of this arm and the one above it, they're more complex than calling `assert_near!` four times.

Reply:
> And it's a public API.

Is it? It's only part of`cfg(test)`...

Reply:
Ah, sorry. Didn't see that in the diff.